earth_enterprise/src/server/gepgcommon

```diff
@@ -0,0 +1,97 @@
+#! /bin/bash
+#
+# Copyright 2017 Open GEE Developers.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+BINDIR=/opt/google/bin
+VARPGDIR=/var/opt/google/pgsql
+GEPGUSER=gepguser
+
+# Runs a command a specific user
+# If it's the same user, then just execute directly, otherwise use su or sudo
+function run_as_user
+{
+  if [ "$( whoami )" == "$1" ]; then
+    >&2 echo "sh -c \"$2\""
+    sh -c "$2"
+  else
+    local use_su=`su $1 -c 'echo -n 1' 2> /dev/null  || echo -n 0`
```

Comment:
Suggestion: This test for `su` will break depending on what shell profile/resource script executes.  It can also get stuck, because `su` waits for the user to enter a password.

Unless `su` gets stuck, you can assume using `su` succeeded. You can use a test like:
```
if su "$1" -c 'echo 1' >/dev/null 2>&1; then
    # something
fi
```